home-assistant-plugin/custom_components/xknx/__init__.py

```diff
@@ -325,11 +325,11 @@ async def telegram_received_cb(self, telegram):
         self.hass.bus.async_fire(
             "knx_event",
             {
-                "data": telegram.payload.value,
+                "data": telegram.payload,
```

Comment:
```suggestion
                "data": telegram.payload.dpt.value,
```
to prevent `WARNING (Recorder) [homeassistant.components.recorder] Event is not JSON serializable: <Event knx_event[L]: data=<GroupValueResponse dpt="<DPTArray value="[0x19,0x62]" />" />, destination=5/1/0, direction=Incoming, source=1.0.54, telegramtype=GroupValueResponse>`
but if a GroupValueRead was received this would not have a `.payload.dpt` attribute...

Reply:
But for a `GroupValueRead`, the data would be empty otherwise. 

I could modify to only set the data property if there is a DPT payload (GroupValueWrite/GroupValueResponse)? There will be other APCI services that don't have a DPT payload (e.g. MemoryRead/Write/Response, PropertyRead/Write/Response, Restart, etc.)

Reply:
`None` is Fine. I have not tested it with a GroupValueRead yet - just want to make sure it doesn't raise AttributeError.

Reply:
Adapted this in b9f2320.